pkg/build/controller/build/defaults/defaults.go

```diff
@@ -187,29 +206,41 @@ func addDefaultEnvVar(build *buildapi.Build, v kapi.EnvVar) {
 }
 
 func addDefaultLabel(defaultLabel buildapi.ImageLabel, buildLabels *[]buildapi.ImageLabel) {
-	found := false
 	for _, lbl := range *buildLabels {
 		if lbl.Name == defaultLabel.Name {
-			found = true
+			return
 		}
 	}
-	if !found {
-		*buildLabels = append(*buildLabels, defaultLabel)
-	}
+	*buildLabels = append(*buildLabels, defaultLabel)
 }
 
-func addDefaultNodeSelector(k, v string, selectors map[string]string) bool {
+func addDefaultNodeSelector(k, v string, selectors map[string]string) {
 	if _, ok := selectors[k]; !ok {
 		selectors[k] = v
-		return true
 	}
-	return false
 }
 
-func addDefaultAnnotations(k, v string, annotations map[string]string) bool {
+func addDefaultAnnotation(k, v string, annotations map[string]string) {
 	if _, ok := annotations[k]; !ok {
 		annotations[k] = v
-		return true
 	}
-	return false
+}
+
+func addDefaultToleration(t kapi.Toleration, tolerations *[]v1.Toleration) error {
+	tol := v1.Toleration{}
+
+	if err := kapiv1.Convert_core_Toleration_To_v1_Toleration(&t, &tol, nil); err != nil {
+		err := fmt.Errorf("Unable to convert core.Toleration to v1.Toleration: %v", err)
+		utilruntime.HandleError(err)
+		return err
+	}
+
+	for _, toleration := range *tolerations {
+		if reflect.DeepEqual(tol, toleration) {
+			return nil
+		}
+	}
+
+	*tolerations = append(*tolerations, tol)
```

Comment:
need to determine if a toleration with the specified key/value/effect already exists, right?

Reply:
@liggitt lines 238-242 are checking for an existing toleration with the same information specified.

Reply:
if a toleration with the same key/value/effect exists with a different toleration seconds, it will fail DeepEqual. what would you expect to happen here?

Reply:
@liggitt There is currently no other way for tolerations to be set on build pods other than through the defaulter, lines 238-242 are a "just in case" type operation.  If/when other ways to set the tolerations become available I would think that this issue would be considered and resolved.